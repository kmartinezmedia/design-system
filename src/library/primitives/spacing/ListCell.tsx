import React, { FunctionComponent, ReactNode } from 'react';
import {
  HStack,
  VStack,
  HStackProps,
  Caption,
  Body,
  Icon,
} from '@designSystem/primitives';
import { noop } from 'lodash';
import { TouchableWithoutFeedback } from 'react-native';
import { GestureResponderEvent } from 'react-native';
import { useAnalytics } from '@utils/analytics/useAnalytics';
import { gutter } from '@designSystem/theme';

export interface ListCellProps extends Omit<HStackProps, 'onPress'> {
  name?: string;
  icon?: ReactNode;
  title: ReactNode;
  subtitle?: ReactNode;
  value?: ReactNode;
  tertiary?: ReactNode;
  ellipsize?: 'fade' | 'tail';
  variant?: 'static' | 'action';
  marker?: 'checked';
  onPress?: (event: GestureResponderEvent) => void;
}

const defaultSpacing = { vertical: 4, horizontal: gutter } as const;

export const ListCell: FunctionComponent<ListCellProps> = ({
  ellipsize = 'tail',
  variant = 'static',
  onPress = noop,
  name,
  icon,
  title,
  subtitle,
  value,
  tertiary,
  marker,
  children,
  spacing = defaultSpacing,
  ...props
}) => {
  const { tapped } = useAnalytics();

  const handlePress = (event: GestureResponderEvent) => {
    tapped(name);
    onPress(event);
  };

  return (
    <TouchableWithoutFeedback onPress={handlePress}>
      <HStack spacing={spacing} alignItems="center" {...props}>
        {icon}
        <VStack
          spacing={icon ? { horizontal: 4 } : 0}
          flexGrow={1}
          flexShrink={1}
          overflow={ellipsize === 'fade' ? 'gradient' : undefined}
        >
          <Body
            ellipsize={ellipsize === 'fade' ? 'clip' : ellipsize}
            spacing={0}
          >
            {title}
          </Body>
          {subtitle ? (
            <Caption ellipsize={ellipsize === 'fade' ? 'clip' : ellipsize}>
              {subtitle}
            </Caption>
          ) : null}
        </VStack>
        {children}
        {value ? (
          <VStack spacing={0} alignItems="flex-end" minWidth="20%">
            <Body spacing={0}>{value}</Body>
            {tertiary ? <Caption>{tertiary}</Caption> : null}
          </VStack>
        ) : null}
        {marker === 'checked' ? (
          <VStack spacing={{ left: 4 }}>
            <Icon size="small" name="check" />
          </VStack>
        ) : null}
        {variant === 'action' ? (
          <VStack spacing={{ left: 4 }}>
            <Icon size="small" name="caretRight" color="muted" />
          </VStack>
        ) : null}
      </HStack>
    </TouchableWithoutFeedback>
  );
};
