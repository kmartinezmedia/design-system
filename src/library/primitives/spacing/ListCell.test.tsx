import React from 'react';
import { View } from 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { ListCell } from './ListCell';

describe('ListCell', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<ListCell title="ðŸ¤”" />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('subtitle', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<ListCell title="ðŸ¤”" subtitle="Sure." />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('value', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <ListCell title="ðŸ¤”" subtitle="Sure." value="$1mil" />,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('tertiary', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <ListCell title="ðŸ¤”" subtitle="Sure." value="$1mil" tertiary="btc" />,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('children', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <ListCell title="ðŸ¤”" subtitle="Sure." value="$1mil" tertiary="btc">
          <View />
        </ListCell>,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('elipsize', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <ListCell
          title="ðŸ¤”"
          subtitle="Sure."
          value="$1mil"
          tertiary="btc"
          ellipsize="fade"
        />,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('variant', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <>
          <ListCell
            title="ðŸ¤”"
            subtitle="Sure."
            value="$1mil"
            tertiary="btc"
            variant="static"
          />
          ,
          <ListCell
            title="ðŸ¤”"
            subtitle="Sure."
            value="$1mil"
            tertiary="btc"
            variant="action"
          />
          ,
        </>,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('marker', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <>
          <ListCell
            title="ðŸ¤”"
            subtitle="Sure."
            value="$1mil"
            tertiary="btc"
            marker="checked"
          />
        </>,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
