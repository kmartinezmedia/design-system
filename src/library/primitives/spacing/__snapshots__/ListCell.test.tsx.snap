// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCell children 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        Sure.
      </Text>
    </View>
    <View />
    <View
      minWidth="20%"
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        $1mil
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        btc
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell default 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell elipsize 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        Sure.
      </Text>
      <BVLinearGradient
        colors={
          Array [
            16777215,
            4294967295,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "bottom": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 30,
          }
        }
      />
    </View>
    <View
      minWidth="20%"
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        $1mil
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        btc
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell marker 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        Sure.
      </Text>
    </View>
    <View
      minWidth="20%"
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        $1mil
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        btc
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          Object {
            "paddingLeft": 16,
          },
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            undefined,
            Object {
              "fontFamily": "CoinbaseIconsSmall",
              "fontSize": 12,
            },
            Object {
              "color": "#1652F0",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell subtitle 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        Sure.
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell tertiary 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        Sure.
      </Text>
    </View>
    <View
      minWidth="20%"
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        $1mil
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        btc
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell value 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "auto",
          "flexDirection": "row",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        🤔
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 16,
            },
            Object {
              "paddingBottom": 4,
            },
            Object {
              "color": "#587088",
              "lineHeight": 16,
              "overflow": "hidden",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        Sure.
      </Text>
    </View>
    <View
      minWidth="20%"
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "auto",
            "flexDirection": "column",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          undefined,
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Graphik-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
            Object {
              "color": "#071422",
              "lineHeight": 24,
              "overflow": "visible",
              "textAlign": undefined,
            },
            undefined,
          ]
        }
      >
        $1mil
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ListCell variant 1`] = `
Array [
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "auto",
            "flexDirection": "row",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          },
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "stretch",
              "alignSelf": "auto",
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "flex-start",
              "width": undefined,
            },
            undefined,
            undefined,
            undefined,
            Object {},
          ]
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              undefined,
              Object {
                "color": "#071422",
                "lineHeight": 24,
                "overflow": "hidden",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          🤔
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 13,
                "fontWeight": "400",
                "lineHeight": 16,
              },
              Object {
                "paddingBottom": 4,
              },
              Object {
                "color": "#587088",
                "lineHeight": 16,
                "overflow": "hidden",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          Sure.
        </Text>
      </View>
      <View
        minWidth="20%"
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "alignSelf": "auto",
              "flexDirection": "column",
              "flexGrow": undefined,
              "flexShrink": undefined,
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "flex-start",
              "width": undefined,
            },
            undefined,
            undefined,
            undefined,
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              undefined,
              Object {
                "color": "#071422",
                "lineHeight": 24,
                "overflow": "visible",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          $1mil
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 13,
                "fontWeight": "400",
                "lineHeight": 16,
              },
              Object {
                "paddingBottom": 4,
              },
              Object {
                "color": "#587088",
                "lineHeight": 16,
                "overflow": "visible",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          btc
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>,
  ",",
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "auto",
            "flexDirection": "row",
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": "nowrap",
            "height": undefined,
            "justifyContent": "flex-start",
            "width": undefined,
          },
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          },
          undefined,
          undefined,
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "stretch",
              "alignSelf": "auto",
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "flex-start",
              "width": undefined,
            },
            undefined,
            undefined,
            undefined,
            Object {},
          ]
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              undefined,
              Object {
                "color": "#071422",
                "lineHeight": 24,
                "overflow": "hidden",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          🤔
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 13,
                "fontWeight": "400",
                "lineHeight": 16,
              },
              Object {
                "paddingBottom": 4,
              },
              Object {
                "color": "#587088",
                "lineHeight": 16,
                "overflow": "hidden",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          Sure.
        </Text>
      </View>
      <View
        minWidth="20%"
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "alignSelf": "auto",
              "flexDirection": "column",
              "flexGrow": undefined,
              "flexShrink": undefined,
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "flex-start",
              "width": undefined,
            },
            undefined,
            undefined,
            undefined,
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              undefined,
              Object {
                "color": "#071422",
                "lineHeight": 24,
                "overflow": "visible",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          $1mil
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Regular",
                "fontSize": 13,
                "fontWeight": "400",
                "lineHeight": 16,
              },
              Object {
                "paddingBottom": 4,
              },
              Object {
                "color": "#587088",
                "lineHeight": 16,
                "overflow": "visible",
                "textAlign": undefined,
              },
              undefined,
            ]
          }
        >
          btc
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "stretch",
              "alignSelf": "auto",
              "flexDirection": "column",
              "flexGrow": undefined,
              "flexShrink": undefined,
              "flexWrap": "nowrap",
              "height": undefined,
              "justifyContent": "flex-start",
              "width": undefined,
            },
            Object {
              "paddingLeft": 16,
            },
            undefined,
            undefined,
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              undefined,
              Object {
                "fontFamily": "CoinbaseIconsSmall",
                "fontSize": 12,
              },
              Object {
                "color": "#587088",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </TouchableWithoutFeedback>,
  ",",
]
`;
